### DISCUSSION TITLE
How to create multiple button in hero card from an array (response from external api)

### DISCUSSION THREAD

**Discussion by tushaar9027** [microsoft/BotFramework-Composer#8936] [Q&A]
Hi All,

If i hard code the button values like this it works

[HeroCard
    title = BotFramework Hero Card
    subtitle = Microsoft Bot Framework
    text = Build and connect intelligent bots to interact with your users naturally wherever they are, from text/sms to Skype, Slack, Office 365 mail and other popular services.
    image = https://sec.ch9.ms/ch9/7ff5/e07cfef0-aa3b-40bb-9baa-7c9ef8ff7ff5/buildreactionbotframework_960.jpg
    buttons = SAP|Windows|Veeva
]

But the API is giving the Array of object and when i use join function on iterable it gives me string which doesn't work properly in buttons field
Output of join 

buttons = "SAP|Windows|Veeva"

does anybody know or has any idea how to achieve this ?

**Comment by smithah** [comment]
# herocardTemplate(image) [HeroCard buttons=${foreach(image, im,cardActionTemplate('messageBack', im.displayText,"{\"msgbck\": \"" + im.qnaid+ "\"}"))} ]

# cardActionTemplate( type, title, value) [CardAction type = ${if(type == null, 'messageBack', type)} Value = ${value} Text = ${title} Title = ${title} displayText = ${title} ]

# heroCards(response) -${herocardTemplate(response)}

Using the above code snippet was able to add multiple buttons on a single HeroCard., here image is an array of json objects. Hope this helps in this case.

**Comment by smithah** [comment]
https://github.com/microsoft/BotFramework-Composer/issues/9523

Had faced a similar usecase.
